misc/package/Data Files/MWSE/core/lib/dependencyManagement/DependencyManager.lua

```diff
@@ -74,8 +78,53 @@ end
 
 
 ---Check if all dependencies are met
----@return boolean #returns true if all dependencies passed, false if any failed.
+---@return boolean passed #returns true if all dependencies passed, false if any failed.
 function DependencyManager:checkDependencies()
+    if not mwse.getConfig("EnableDependencyChecks") then
+        return true
+    end
+
+    -- Don't check dependencies if a mod't plugin is not active.
+    -- Log if there is metadata.toml pointing to a partially installed/uninstalled mod.
+    local plugin = self.metadata.package.plugin
+    local pluginExists = plugin and util.pluginExists(plugin)
+    local pluginDisabled = pluginExists and not tes3.isModActive(plugin)
+
+    local luaMod = self.metadata.tools and self.metadata.tools.mwse and self.metadata.tools.mwse["lua-mod"]
+    local luaModExists = luaMod and util.luaModExists(luaMod)
+
+    local luaOnly = luaMod and not plugin or false
+    local pluginOnly = plugin and not luaMod or false
+    local luaPlugin = luaMod and plugin and true or false
+
+    local uncomplete = (luaOnly and not luaModExists) or
+                       (pluginOnly and not pluginExists) or
+                       (luaPlugin and not (luaModExists and pluginExists))
+
+    if uncomplete then
```

Comment:
Took me a while to figure out what's actually happening here. I think it can be simplified a lot:

```lua
    local plugin = self.metadata.package.plugin
    local pluginExists = plugin and util.pluginExists(plugin)
    local pluginDisabled = pluginExists and not tes3.isModActive(plugin)
    
    local luaMod = self.metadata.tools and self.metadata.tools.mwse and self.metadata.tools.mwse["lua-mod"]
    local luaModExists = luaMod and util.luaModExists(luaMod)

    -- Check if either an expected luaMod or plugin is missing
    local incomplete = false
    if luaMod and not luaModExists then
        incomplete = true
    elseif plugin and not pluginExists then
        incomplete = true
    end

    if incomplete then
```

Reply:
Yeah, my initial solution is messy. Your proposal works identically and is much easier to follow. Applied.